include/nbl/builtin/hlsl/bxdf/fresnel.hlsl

```diff
@@ -17,55 +17,110 @@ namespace hlsl
 namespace bxdf
 {
 
-namespace impl
+namespace fresnel
 {
-template<typename T>
-struct orientedEtas;
 
-template<>
-struct orientedEtas<float>
+template<typename T NBL_PRIMARY_REQUIRES(is_scalar_v<T> || is_vector_v<T>)
```

Comment:
you can require `FloatingPointLikeVectorial<T>` should cover all your bases

Reply:
`FloatingPointLikeVectorial<T>` doesn't cover scalars. Do we only want vectors for orientedEtas?

Reply:
continued in https://github.com/Devsh-Graphics-Programming/Nabla/pull/899/files?diff=unified&w=1#r2183770288